internal/engine/execution/procedure.go

```diff
@@ -467,6 +468,24 @@ func makeExecutables(params []*generate.InlineExpression) []evaluatable {
 // a slice of values in the order of the keys. If a value can be converted to an
 // int, it will be. If a value does not exist, it will be set to nil.
 func orderAndCleanValueMap(values map[string]any, keys []string) []any {
+	// we need to iterate over all values, and see if it has a $. If not,
+	// we need to add one so that the keys (which have $) match the values
+	// (which do not have $)
+	cloned := false
+	for k, v := range values {
+		if k[0] != '$' {
```

Comment:
Any need to worry about contextual variables (`@` prefixed) here?
Would it be safer to deal with this in the `range keys` loop below, with a `!ok` condition handled by `values[strings.TrimPrefix(key, "$")]` or something similar?

Reply:
Good question. We actually don't pass `@` prefixed variables via values anymore. We did in SQLite because you can bind `@` variables in a statement, but internally we translate them to `current_setting`, so it doesn't really matter.

If `@caller` comes in, it will get translated to `$@caller`, but it would get ignored anyways.

Reply:
> but internally we translate them to `current_setting`, so it doesn't really matter.

Ah right.  I had my head in the past from the fractal thread, but they are on postresql with 0.7 too.

> If `@caller` comes in, it will get translated to `$@caller`, but it would get ignored anyways.

Even in the context of `(*dmlStmt).execute` where it's not an ad hoc query?  With an action, the keys passed would be from:

https://github.com/kwilteam/kwil-db/blob/4ea9379752af786bb6c310ad2c06e7be89326bfb/internal/engine/generate/actions.go#L51-L55

Is the above comment still accurate?  I think it is.

`orderAndCleanValueMap` is a simple utility function, so context is not obvious.

Reply:
If `delete(values, k)` were removed below, we wouldn't lose the ability to lookup the contextual variable, if I'm reading this right.

Reply:
Correct, however I'm not planning on backporting this, as I don't really think it is a bug, as much as a UX improvement

Reply:
Yah, I agree.  Want to merge this now or are you planning for the ParseSQL fix here?

Reply:
I'll make the ParseSQL fix in a separate PR since we will want to backport it

Reply:
Let me update the docs for `orderAndCleanValueMap` though

Reply:
Oh, so should `(*ProcedureContext) Values() map[string]any` stop setting the contextual variables?  That part of the code is a bit misleading now.  The SET LOCAL / current_setting approach is used for both actions and procs, right?  I guess it could still be helpful for debugging.

Reply:
It probably should. I considered this, but avoided it b/c it is a breaking change, and then forgot to get rid of it for v0.8. But yeah probably.